Rendera/include/cube.hpp

```diff
@@ -7,6 +7,23 @@
 #include "ray.hpp"
 #include "material.hpp"
 
+
+//debug
+class HitInfo{
+    bool if_hit;
+    vec3 dist_v;
+    Material mat;
+    ray r;
+    int depth;
+
+    public:
+        HitInfo(bool if_hit, vec3 v, Material m, ray r, int d):if_hit{if_hit}, dist_v{v}, mat{m}, r{r}, depth{d}{};
```

Comment:
Members of `struct`s are public by default. Also, this constructor is not needed. The compiler will auto-generate it for you.